Adding a reference test where a bigger area than the original element

Adenilson opened the pull request:
Adding a reference test where a bigger area than the original element size is defined at script.

To pass, this depends in a fix on rust-azure (i.e. #195). No for commit, we need the rust-azure to land and next update the deps.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/servo/servo/7414)

<!-- Reviewable:end -->

Adenilson commented:
Blocked on this landing: https://github.com/servo/rust-azure/pull/195

Adenilson commented:
Not for landing yet, as there are some pending questions on: https://github.com/servo/rust-azure/pull/195 and https://github.com/servo/rust-azure/issues/196

pcwalton commented:
@bors-servo: r+

bors-servo commented:
:pushpin: Commit 89abdce has been approved by `pcwalton`

<!-- @bors-servo r=pcwalton 89abdce29a94055b3d91e55294b26197856498ee -->

bors-servo commented:
:hourglass: Testing commit 89abdce with merge 5a0cf62...

bors-servo commented:
:broken_heart: Test failed - [linux1](http://build.servo.org/builders/linux1/builds/2919)

larsbergstrom commented:
@bors-servo retry

(can't read the failure logs anymore due to the machine rename?)

bors-servo commented:
:hourglass: Testing commit 89abdce with merge 0cf63d3...

bors-servo commented:
:broken_heart: Test failed - [linux-dev](http://build.servo.org/builders/linux-dev/builds/1)

jdm commented:
```
command timed out: 1200 seconds without output running ['./mach', 'test-ref', '--kind', 'cpu'], attempting to kill
```

Adenilson commented:
@jdm any idea why the new test never gets executed? (i.e. canvas_over_area.html)

Using a linux laptop (i7@2.9Ghz), all the ref tests runs just fine with this patch applied:
https://gist.github.com/Adenilson/d5ec1027ead6ec193309

And I got the same result on OSX. 

Could it be related to the upgrade on rust-azure?

bors-servo commented:
:umbrella: The latest upstream changes (presumably #7605) made this pull request unmergeable. Please resolve the merge conflicts.

Manishearth commented:
I rebased this and moved it to the wpt tests. It times out on my machine.

https://github.com/servo/servo/compare/master...Manishearth:pr-7414?expand=1

Adenilson commented:
@Manishearth thanks for porting this test to wpt. I will have a look and test locally to see what is the behavior.

Adenilson committed:
Adding a reference test where a bigger area than the original
element is defined in script.

The test is now using wpt infrastructure (thanks Manishearth!).

Adenilson commented:
This is how servo currently renders both the test and the ref: 
<img width="326" alt="screen shot 2015-11-02 at 8 43 07 am" src="https://cloud.githubusercontent.com/assets/959744/10888126/4f328308-813f-11e5-95e5-4a418733ecb3.png">

Manishearth commented:
@bors-servo r=pcwalton

bors-servo commented:
:pushpin: Commit 4fdf73d has been approved by `pcwalton`

<!-- @bors-servo r=pcwalton 4fdf73d1055a31f7acef69fa1152615b6aa8e840 -->

jdm commented:
@bors-servo: r-

jdm commented:
Missing `<link>` to make the harness know what to do: http://testthewebforward.org/docs/reftests.html#components-of-a-reftest
This is why modifying MANIFEST.json manually is always the wrong thing to do :)

Manishearth commented:
I thought I had done that?

Manishearth commented:
Oh, it didn't get carried over into the PR, and I didn't notice.

Adenilson committed:
Telling the test runner what compare the test with.

Adenilson committed:
The end of the MANIFEST odyssey.

Adenilson commented:
Now it should be tidy and clean.
 
Summary
=======

Ran 1 tests
Expected results: 1
Unexpected results: 0

OK

pcwalton commented:
@bors-servo: r+

bors-servo commented:
:pushpin: Commit b1197c3 has been approved by `pcwalton`

<!-- @bors-servo r=pcwalton b1197c31d1bff0192d0e25c1a65e21bd3c76c0f6 -->

bors-servo commented:
:hourglass: Testing commit b1197c3 with merge db1163b...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/163), [gonk](http://build.servo.org/builders/gonk/builds/168), [linux-dev](http://build.servo.org/builders/linux-dev/builds/161), [linux-rel](http://build.servo.org/builders/linux-rel/builds/175), [mac-dev-ref-unit](http://build.servo.org/builders/mac-dev-ref-unit/builds/174), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/173), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/183)
